csm/core/controllers/maintenance.py

```diff
@@ -66,6 +67,12 @@ async def post(self):
         action = self.request.match_info[const.ACTION]
         body = await self.request.json()
         body[const.ACTION] = action
+
+        #if hostname is obtained in request body then get the nodeid mapped to the hostname.
+        hostname_nodeid_map = Conf.get(const.CSM_GLOBAL_INDEX, f"{const.MAINTENANCE}")
+        rev_hostname_nodeid_map = {host:node for node, host in hostname_nodeid_map.items()}
+        if rev_hostname_nodeid_map.get(body[const.RESOURCE_NAME]):
+            body[const.RESOURCE_NAME] = rev_hostname_nodeid_map.get(body[const.RESOURCE_NAME])
```

Comment:
where is the else condition? If resource is not found in nodeid_map then what should be the action? An exception should be thrown?

Reply:
`rev_hostname_nodeid_map = {"sm17-r18":"srvnode-1", "sm18-r18": "srvnode-2"}`

if body[const.RESOURCE_NAME] is hostname and present in  rev_hostname_nodeid_map then its corresponding nodeid is taken for further process.
Else nodeid obtained from user goes into same validation process as before.

Reply:
but you will get hostname from the UI now, not the node id.

Reply:
Yes correct. Thats why srvnode id corresponding to hostname is obtained from conf. 
And this may not be the case from CLI. We can get hostname or nodeid from cli cmd